src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -37146,7 +37295,7 @@ static bool TryGetValue<T>([AllowNull]T x, [MaybeNullWhen(true)]out T y, [MaybeN
     {
         y = x;
         z = x;
-        return y != null || z != null; // 1, 2
+        return y != null || z != null;
```

Comment:
Why would this not have warnings? There should be different state here between the branches? Or is it because it's nested inside a `||`?

Reply:
The when-true and when-false states for the `return` are identical (every variable may be null), so the expression as a whole is not considered to inform the nullability of `y` or `z`.

---
In reply to: [503605598](https://github.com/dotnet/roslyn/pull/48425#discussion_r503605598) [](ancestors = 503605598)